### PR TITLE
Iruleuri.tcl update to URI::compare, URI::encode and URI::decode

### PR DESCRIPTION
**Pull request by apextc**
Update to `URI::encode` and `URI::decode` to use string map.

`URI::encode` has been added to `%7e` / `~` excluding letter, number and some select symbols.
`URI::decode` decodes all characters up to `%7e` / `~` and string map used to map both upper and lower case charaters. 

e.g. `%2b` and `%2B` map to `+` using `string map {"%2b" "+" "%2B" "+"}`

Reference using UTF-8 from [HTML URL Encoding Reference](https://www.w3schools.com/tags/ref_urlencode.asp)

### CODE REVIEW

**Comment by landro** [comment]
Nice work, @apextc ! I'll try to review and merge this within the next few days!

File: src/iruleuri.tcl [hunk 1]
```diff
@@ -22,183 +22,148 @@ namespace eval ::testcl::URI {
...
-        } elseif { "%5d" eq $char } {
-          log::log debug " - Converting encoded char $char"
-          append decodedString "]"
-        } else {
-          log::log critical "Not converting char $char - to be implemented in iruleuri.tcl"
-          append decodedString "$char"
-        }
-     }
-  }
+  set utf8_uri [encoding convertto utf-8 $uri]
```
Comment (apextc) [inline src/iruleuri.tcl]: This line is not needed, left in error

Comment (apextc) [reply src/iruleuri.tcl]: I have removed

**Comment by apextc** [comment]
@landro I have an update to the URI::compare as well, do you want it added to this pull request or would you prefer me to create a new one so you can review independently?

If you want to quickly take a look - [apextc/TesTcl commit 'rewrite testcl::URI::compare'](https://github.com/apextc/TesTcl/commit/ff8638ed342460ed115e76317c3595cc27883688)

**Comment by landro** [comment]
Same PR is Ok

Sendt fra min iPhone

**Comment by landro** [comment]
Great work @apextc !

**Pull request merged**